src/main/java/hu/wurfel/refference/school/diary/DiaryCrudService.java

```diff
@@ -19,47 +19,47 @@ public List<Diary> getAllDiaries() {
         return new ArrayList<>(diaryRepository.findAll());
     }
 
-    public Diary getDiaryByDiaryid(Integer Id) {
+    public Diary getByDiaryid(Integer Id) {
         return diaryRepository.findById(Id).get();
     }
 
-    public List<Diary> getDiariesByScid(Long scid) {
+    public List<Diary> getByScid(Long scid) {
         return diaryRepository.findAllByStudentId(scid);
     }
 
-    public List<Diary> getDiariesByClassid(Integer cid) {
+    public List<Diary> getByClassid(Integer cid) {
         return diaryRepository.findAllByClassID(cid);
     }
 
-    public Diary getDiaryByScidAndClassid(Long scid, Integer cid) {
+    public Diary getByScidAndClassid(Long scid, Integer cid) {
         return diaryRepository.findByStudentIdAndClassID(scid, cid).get();
     }
 
-    public Diary saveDiary(@NotNull Diary diary) {
+    public Diary save(@NotNull Diary diary) {
         diaryRepository.save(diary);
-        return getDiaryByDiaryid(diary.getId());
+        return getByDiaryid(diary.getId());
     }
 
-    public Diary saveDiary(Integer did, Long scid, Integer cid) {
-        return saveDiary(new Diary(did, scid, cid));
+    public Diary save(Integer did, Long scid, Integer cid) {
+        return save(new Diary(did, scid, cid));
```

Comment:
Nagyon nem célszerű kézzel megadni az id-t és úgy menteni a db-be. Ha itt valami speckó, nem generált adat kell ID-nak, akkor oké, viszont ha mindkét módszer használod, az könyen beríthat dolgokat. Ezért is mondtam a Studentnél, hogy a card number legyen simán egy unique field, de az ID maradjon db id.

Reply:
Az ID-s mentést csak új létrehozásánal használom

Reply:
minden más esetben dbből lekértet módosítok és save-elem  vissaz az entity-t váróval

Reply:
*

Reply:
Ennél meg majd beszéljünk a tranzakciókezelésről.